fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -1512,432 +1526,477 @@ private Bundle processBundleEntries(Bundle requestBundle, Bundle responseBundle,
-
-                        } else if (pathTokens.length == 2 && "_search".equals(pathTokens[1])) {
-                            // This is a 'search' request.
-                            Bundle searchResults =
-                                    doSearch(pathTokens[0], null, null, queryParams, absoluteUri, null, null);
-
-                            // Save the results of the operation in the bundle response field.
-                            Bundle.Entry.Response.Builder responseBuilder = response.toBuilder();
-                            responseBuilder.status(string(Integer.toString(SC_OK)));
-
-                            responseIndexAndEntries.put(entryIndex, responseEntryBuilder
-                                    .resource(searchResults)
-                                    .response(responseBuilder.build())
-                                    .build());
-
-                            setBundleResponseStatus(response, SC_OK, requestDescription.toString(), initialTime);
-                        } else if (pathTokens.length == 1) {
-                            // This is a 'create' request.
-
-                            // Retrieve the local identifier from the request entry (if present).
-                            String localIdentifier =
-                                    retrieveLocalIdentifier(requestEntry, localRefMap);
-
-                            // Retrieve the resource from the request entry.
-                            Resource resource = requestEntry.getResource();
-                            if (resource == null) {
-                                String msg =
-                                        "BundleEntry.resource is required for bundled create requests.";
-                                throw buildRestException(msg, IssueType.NOT_FOUND);
-                            }
-
-                            // Convert any local references found within the resource to their
-                            // corresponding external reference.
+                        responseBuilder.status(string(Integer.toString(SC_NOT_FOUND)));
+                        responseIndexAndEntries.put(entryIndex, responseEntryBuilder.resource(FHIRUtil.buildOperationOutcome(e, false)).response(responseBuilder.build()).build());
+                        logBundleRequestCompletedMsg(requestDescription.toString(), initialTime, SC_NOT_FOUND);
+                    } catch (FHIRPersistenceResourceDeletedException e) {
+                        if (failFast) {
+                            String msg = "Error while processing request bundle.";
+                            throw new FHIRRestBundledRequestException(msg).withIssue(e.getIssues());
+                        }
+                        
+                        Bundle.Entry.Response.Builder responseBuilder = response.toBuilder();
+                        responseBuilder.status(string(Integer.toString(SC_GONE)));
```

Comment:
since this if fixed, it may be great to make that string value a static final string(Integer.toString(SC_GONE))

Reply:
Done.